src/pay/utils/handlePayRequest.ts

```diff
@@ -0,0 +1,27 @@
+import { buildPayTransaction } from '../../api';
+import type { BuildPayTransactionParams } from '../../api';
+import type { HandlePayRequestParams } from '../types';
+
+export const handlePayRequest = async ({
+  address,
+  chargeHandler,
+  productId,
+}: HandlePayRequestParams) => {
+  const buildPayTransactionParams: BuildPayTransactionParams = {
+    address,
+  };
+
+  if (chargeHandler) {
+    buildPayTransactionParams.chargeId = await chargeHandler();
+  } else if (productId) {
+    buildPayTransactionParams.productId = productId;
+  }
```

Comment:
Can we add a small comment here?

Reply:
will add the comment on the React component and not this util function

Reply:
https://github.com/coinbase/onchainkit/pull/1338/files#diff-b7dd4bc7cc964af915e429bc56534f129be883d8ace8ace0523cc298b167965dR22-R24